mysql-compatibility: add doc for a UPDATE incompatibility case (#7094)
pingcap/docs: mysql-compatibility.md

- The syntax `SELECT ... GROUP BY ... WITH ROLLUP` is not supported.
- The syntax `SELECT .. GROUP BY expr` does not imply `GROUP BY expr ORDER BY expr` as it does in MySQL 5.7. TiDB instead matches the behavior of MySQL 8.0 and does not imply a default order.

> ### Update statement

Reviewer: ```suggestion
### `UPDATE` statement
```

> 

Reviewer: ```suggestion

Taking the following statements as an example.

```

```sql
CREATE TABLE companies (id bigint primary key, ida bigint);
INSERT INTO companies VALUES (14, 14);
UPDATE companies SET id = id + 1, ida = id * 2;
```

> In MySQL the result is

Reviewer: ```suggestion
In MySQL, the result is as follows:
```

```sql
mysql> select * from companies;
+----+------+
| id | ida  |
+----+------+
| 15 |   30 |
+----+------+
1 row in set (0.00 sec)
```

> In TiDB the result is

Reviewer: ```suggestion
In TiDB, the result is as follows:
```

```sql
mysql> select * from companies;
+----+------+
| id | ida  |
+----+------+
| 15 |   28 |
+----+------+
1 row in set (0.00 sec)
```

> In TiDB, `id` is always the original value of the `id` column, while in MySQL, the value of the later `id` is result of the `id` after the evaluation of the previous expression.

Reviewer: ```suggestion
In TiDB, `id` is always the original value of the `id` column, which is `14` in the above example.

In MySQL, the value of the later `id` (which is the `id` in `ida = id * 2`) is result of the `id` after the evaluation of the previous expression (which is `id = id + 1` in the above example).
```
